packages/ipfs-core/src/utils.js

```diff
@@ -96,36 +96,45 @@ const resolvePath = async function (ipld, ipfsPath, options = {}) {
  * @param {boolean} [options.includeContent]
  */
 const mapFile = (file, options = {}) => {
+  if (file.type !== 'file' && file.type !== 'directory' && file.type !== 'raw') {
+    // file.type === object | identity not supported yet
+    throw new Error(`Unknown node type '${file.type}'`)
+  }
+
   /** @type {import('ipfs-core-types/src/root').IPFSEntry} */
   const output = {
     cid: file.cid,
     path: file.path,
     name: file.name,
     depth: file.path.split('/').length,
-    size: 0,
+    size: file.size,
```

Comment:
is it legit to pass through the raw size for a raw node?

Reply:
Yes, it'll be the [size of the block](https://github.com/ipfs/js-ipfs-unixfs/blob/master/packages/ipfs-unixfs-exporter/src/resolvers/raw.js#L48)